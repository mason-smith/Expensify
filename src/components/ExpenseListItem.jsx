import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import Divider from "@material-ui/core/Divider";

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <ListItem dense component={Link} button to={`/edit/${id}`}>
      {/* <Link to={`/edit/${id}`}> */}
      <ListItemText
        primary={description}
        secondary={moment(createdAt).format("MMMM Do, YYYY")}
      />{" "}
      {/* </Link> */}
      <p> {numeral(amount / 100).format("$0,0.00")}</p>
    </ListItem>
    <Divider />
  </div>
);

export default ExpenseListItem;
