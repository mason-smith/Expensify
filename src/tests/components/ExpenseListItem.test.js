import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem.jsx';
import expenses from '../fixtures/expenses.js';

test('should render ExpenseListItem with expenses', () => {
 const wrapper = shallow(<ExpenseListItem expenses={expenses} />);
 expect(wrapper).toMatchSnapshot();
});
