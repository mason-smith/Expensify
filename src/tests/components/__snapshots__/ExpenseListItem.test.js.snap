// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with expenses 1`] = `
<div>
  <h3>
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      exact={true}
      to="/edit/undefined"
    >
        
       
    </NavLink>
  </h3>
  <p>
     
     - 
     
  </p>
</div>
`;
